### PR TITLE
make subnet names lowercase

### LINKED ISSUE #459: make subnet names lowercase
Right now we have `vpc.subnets.{Public,Private}`, these should be lowercase. We probably want to move away for using a map and use a struct actually.

### PR DESCRIPTION
**Pull request by errordeveloper**
### Description

Fixes #459.

This changes subnets from `map[SubnetTopology]map[string]Network` to a struct with `Private` and `Public` fields. That is in order to make those lowercase in JSON/YAML, so that it's consistent with the rest of the API.

### Checklist
- [x] Code compiles correctly (i.e `make build`)
- [x] All tests passing (i.e. `make test`)
- [x] Added/modified documentation as required (such as the README)

### CODE REVIEW

**Comment by errordeveloper** [comment]
I actually want to make it fail when old capitalised names are used, but I'm not too sure if I must. Perhaps we can try using `gjson.Get(configData, "vpc.subnets.Private")`, but that means we need turn YAML into JSON first... I'm undecided.

**Comment by jrryjcksn** [comment]
Thanks for giving be the background! The more I can learn sooner, the better. :-)

File: pkg/cfn/manager/nodegroup.go [hunk 1]
```diff
@@ -223,7 +223,7 @@ func (c *StackCollection) mapStackToNodeGroupSummary(stack *Stack) (*NodeGroupSu
 		return nil, errors.Wrapf(err, "error getting Cloudformation template for stack %s", *stack.StackName)
 	}
 
-	cluster := getClusterName(stack)
+	cluster := getClusterNameTag(stack)
```
Comment (mumoshu) [inline pkg/cfn/manager/nodegroup.go]: Doesn't this revive #510? 

`stack` here is a nodegroup stack whose name is `<cluster name>-nodegroup-<nodegroup name>`, but then `getClusterNameTag` will end up returning a empty string.

Comment (errordeveloper) [reply pkg/cfn/manager/nodegroup.go]: No, this will get just the tag, and nodegroup stack has cluster name tag.

Comment (mumoshu) [reply pkg/cfn/manager/nodegroup.go]: Seems like I'm not following your idea correctly. Anyway, running eksctl built from your branch(at 2bc7cef2f793025eca9211ffc9b1d0a19f63ff93) produces empty cluster names in `get nodegroup`:

```
$ ./eksctl get nodegroup --cluster mycluser
CLUSTER	NODEGROUP	CREATED			MIN SIZE	MAX SIZE	DESIRED CAPACITY	INSTANCE TYPE	IMAGE ID
	nodegroup1	2019-02-02T13:06:24Z	2		2		0			t3.large	ami-0f0e8066383e7a2cb
```

Comment (mumoshu) [reply pkg/cfn/manager/nodegroup.go]: Probably you wanted a dedicated `getClusterName` func for nodegroups?

**Comment by errordeveloper** [comment]
Ah, so looks like I should make a separate PR to address this issue, not as
trivial as I hoped!

File: pkg/cfn/manager/nodegroup.go (hunk 1 above)
Comment (errordeveloper) [reply pkg/cfn/manager/nodegroup.go]: Ah, so looks like I should make a separate PR to address this issue, not as trivial as I hoped!

> Probably you wanted a dedicated `getClusterName` func for nodegroups?

Not as such, `getClusterName` was intended to be sufficient for this particular case.

File: pkg/cfn/manager/cluster.go [hunk 2]
```diff
@@ -161,3 +157,14 @@ func getClusterName(s *Stack) string {
 	}
 	return ""
 }
+
+func getClusterNameTag(s *Stack) string {
+	for _, tag := range s.Tags {
+		if *tag.Key == api.ClusterNameTag {
+			if strings.HasSuffix(*s.StackName, "-cluster") {
```
Comment (errordeveloper) [inline pkg/cfn/manager/cluster.go]: Should have removed this suffix check, that's all. Will open another PR, so that we keep this PR nicely scoped also.

**Comment by mumoshu** [comment]
Makes sense! Thanks for confirming.
2019年2月7日(木) 18:18 Ilya Dmitrichenko <notifications@github.com>:

**Comment by errordeveloper** [comment]
@mumoshu https://github.com/weaveworks/eksctl/pull/517

**Comment by errordeveloper** [comment]
I don't get why, but somehow this has affected integration tests and I have fix coming up in another PR.

**Pull request merged**